lib/spack/spack/ci.py

```diff
@@ -775,17 +776,33 @@ def generate_gitlab_ci_yaml(
             "instead.",
         )
 
-    if "mirrors" not in yaml_root or len(yaml_root["mirrors"].values()) < 1:
-        tty.die("spack ci generate requires an env containing a mirror")
+    pipeline_mirrors = spack.mirror.MirrorCollection(binary=True)
+    deprecated_mirror_config = False
+    buildcache_destination = None
+    if "buildcache-destination" in pipeline_mirrors:
+        buildcache_destination = pipeline_mirrors["buildcache-destination"]
+    else:
+        deprecated_mirror_config = True
+        # TODO: This will be an error in Spack 0.23
+
+    # TODO: Remove this block in spack 0.23
+    remote_mirror_url = None
+    if deprecated_mirror_config:
+        if "mirrors" not in yaml_root or len(yaml_root["mirrors"].values()) < 1:
+            tty.die("spack ci generate requires an env containing a mirror")
 
-    ci_mirrors = yaml_root["mirrors"]
-    mirror_urls = [url for url in ci_mirrors.values()]
-    remote_mirror_url = mirror_urls[0]
+        ci_mirrors = yaml_root["mirrors"]
+        mirror_urls = [url for url in ci_mirrors.values()]
+        remote_mirror_url = mirror_urls[0]
 
     spack_buildcache_copy = os.environ.get("SPACK_COPY_BUILDCACHE", None)
```

Comment:
Should this still be guarded by the environment variable?

Reply:
I don't follow, which environment variable should be guarding what?

Reply:
The existence of `SPACK_COPY_BUILDCACHE` is still the activation requirement for writing the manifest file. The new method uses known named mirrors to determine the copy location. I think that this should be fixed.

Reply:
I think you might be confusing copy-only pipelines (where we now use known named mirrors to know the copy destination), with this env var, which is used to trigger writing the manifest file.

Reply:
Ah, right, I am merging our other conversion about the manifest file into this PR. We can address this later.